packages/babel-preset/common.js

```diff
@@ -0,0 +1,100 @@
+const nonStandardPlugins = require('./non-standard-plugins');
+
+module.exports = function shopifyCommonPreset(api, options = {}) {
+  const {
+    targets = {
+      node: 'current',
+    },
+    corejs = 3,
+    debug = false,
+    modules = 'auto',
+    typescript = false,
+    useBuiltIns = 'entry',
+    transformRuntime = false,
+    transformRuntimeOptions = {
+      corejs: false,
+      helpers: true,
+      // By default, babel assumes babel/runtime version 7.0.0-beta.0,
+      // explicitly resolving to match the provided helper functions.
+      // https://github.com/babel/babel/issues/10261
+      version: require('@babel/runtime/package.json').version,
+      regenerator: true,
+      // https://babeljs.io/docs/en/babel-plugin-transform-runtime#useesmodules
+      // We should turn this on once the lowest version of Node LTS
+      // supports ES Modules.
+      useESModules: false,
+      // This allows users to run transform-runtime broadly across a whole project.
+      // By default, transform-runtime imports from @babel/runtime/foo directly, but
+      // that only works if @babel/runtime is in the node_modules of the file that is being compiled.
+      absoluteRuntime: false,
+    },
+  } = options;
+
+  const isNode = 'node' in targets;
+
+  const presets = [
+    [
+      require.resolve('@babel/preset-env'),
+      {
+        modules,
+        useBuiltIns,
+        corejs,
+        targets,
+        debug,
+        bugfixes: true,
+      },
+    ],
+  ];
+
+  const plugins = [
+    ...nonStandardPlugins(options),
+    isNode && require.resolve('@babel/plugin-proposal-dynamic-import'),
+    isNode && require.resolve('@babel/plugin-transform-modules-commonjs'),
+    typescript && require.resolve('@babel/preset-typescript'),
+    // polyfill for import statements in webpack for browsers that don't support promises
+    require.resolve('@babel/plugin-syntax-dynamic-import'),
```

Comment:
Is this still useful? Don't we support browsers that support promises?

Reply:
Good point. It most likely isn't useful anymore.

Reply:
I think `@babel/plugin-proposal-dynamic-import`, `@babel/plugin-syntax-dynamic-import` and `@babel/plugin-transform-modules-commonjs` are all provided by preset-env in 7.13 (probably added in earlier versions) so we don't need to explicitly state them anymore.


Using blame in https://github.com/babel/babel/blob/main/packages/babel-preset-env/src/available-plugins.js be useful for researching this